### DISCUSSION TITLE
Install of Podman for all users on a Windows system

### DISCUSSION THREAD

**Discussion by bhunsicker** [containers/podman#14942] [Q&A]
Hi Everyone,

I am trying to set up an installer for our enterprise.  Here are the steps I have thus far:

Prep Work:
Pull down the Fedora image and put in the resolv.conf and wsl.conf files along with the needed certs
Load Podman using the `podman machine init` on a segregated machine
zip up the binaries from c:\program files \RedHat\Podman
zip up the user files from c:\Users\<MyUser>
- ~\.ssh
- ~\.ssh\podman-machine-default
- ~\.ssh\podman-machine-default.pub
- ~\.local\share\containers\podman\machine\wsl\podman-machine-default
- ~\.local\share\containers\podman\machine\wsl\wsldist\podman-machine-default
- ~\.local\share\containers\podman\machine\wsl\wsldist\podman-machine-default\ext4.vhdx (NOTE:  This file may not be needed)
- ~\.config\containers\podman\machine\wsl
- ~\.config\containers\podman\machine\wsl\podman-machine-default.json (NOTE: modify this file to point to the Public folder rather than my user)
- ~\AppData\Roaming\containers
- ~\AppData\Roaming\containers\containers.conf
extract the WSL image from podman-machine-default

Install on a given endpoint:
Extract the Podman Binaries to c:\program files \RedHat\Podman
Set the path to add the folder
Extract the Podman user zip to the c:\Users\Public folder :
Import the exported WSL2 image

At this point I can start the WSL2 image using `wsl -d podman-machine-default` and it starts as expected.  I am not able to run podman machine start as it does not find the expected VM
```
> podman machine start
Error: podman-machine-default: VM does not exist
```
If I were to extract the Podman user zip into the same user I used to create the files, then everything seems to work fine for that user, but other users dont seem to be able to see the VM as it is owned by the original install user.

If anyone has some advice on how to go about making the install usable for all users on the Windows system, that would be appreciated.

**Comment by mheon** [comment]
@n1hility PTAL

**Comment by bhunsicker** [comment]
Ok, so more on this issue.  Doing the podman system connection list does not return anything (assuming because the connection info is being looked for in ~\AppData\Roaming\containers\containers.conf.  That would probably explain why other processes are not working as expected.

is there a supported process for making podman globally accessible for all users on a system?

**Comment by n1hility** [comment]
The way WSL works, the VM is isolated per user, so by extension each user has to create their own podman machine instance. Technically you could run the podman-machine distro on one user, and access it via podman machine ssh on another. However, direct wsl access via the wsl command by other users would not be possible.

**Comment by bhunsicker** [comment]
OK, Thank you again.

**Comment by gbraad** [comment]
Alternatively, the use of CRC as a VM might help in this regard. At the moment we prevent non-administrator installs, and make them user- specific. But that is because we experienced issues when people installed as an Administrator and then were unable to locate their instance, as they were impossible to find/access as a user.
